control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go

```diff
@@ -45,7 +45,16 @@ func ReconcileNetworkOperator(network *operatorv1.Network, networkType hyperv1.N
 				network.Spec.DefaultNetwork.OVNKubernetesConfig.GenevePort = &port
 			}
 		}
+	case hyperv1.PowerVSPlatform:
+		if networkType == hyperv1.OVNKubernetes {
+			if network.Spec.DefaultNetwork.OVNKubernetesConfig == nil {
+				network.Spec.DefaultNetwork.OVNKubernetesConfig = &operatorv1.OVNKubernetesConfig{}
```

Comment:
why is this needed?

Reply:
Followed the same pattern from KubeVirt platform https://github.com/openshift/hypershift/blob/main/control-plane-operator/hostedclusterconfigoperator/controllers/resources/network/reconcile.go#L41

Initially `network.Spec.DefaultNetwork.OVNKubernetesConfig` is nil and tried to initialise with empty struct and again initialled `GatewayConfig` to set `RoutingViaHost` to true

Reply:
To align with the other part of the block of code refactored slightly.
@enxebre Kept `GatewayConfig` reconciliation inside nil check or else it would try to reassign the same obj on every reconcilation. Please let me know whether that answers your question?